File: 백준/Silver/21600. 계단/계단.py
n = int(input())
histogram = list(map(int, input().split()))
lengthOfStair = 0
result = 0
for x in histogram:
    lengthOfStair = lengthOfStair+1 if lengthOfStair+1 <= x else x
    result = max(result, lengthOfStair)
print(result)

File: 백준/Silver/21600. 계단/계단.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];
let iter = 0;
rl.on('line', (line) => {
  input.push(line);
}).on('close', () => {
  const N = Number(input[iter++]);
  const l = [...input[iter++].split(' ')].map((item) => Number(item));
  let last = 1;
  let answer = last;
  for (let i = 0; i < N; i++) {
    if (l[i] >= last) {
      last += 1;
      answer = Math.max(answer, last);
    } else last = l[i];
  }
  answer = Math.max(answer, last);
  console.log(answer);
  process.exit();
});
